### PR TITLE
Block data migration

### PR DESCRIPTION
**Pull request by alecps**
Ports https://github.com/celo-org/optimism/pull/121 over to celo6

### CODE REVIEW

File: .github/workflows/docker-build-scan.yaml [hunk 1]
```diff
@@ -1,92 +1,50 @@
 name: Docker Build Scan
 on:
+  pull_request:
+    branches:
```
Comment (alecps) [inline .github/workflows/docker-build-scan.yaml:4]: These changes added here https://github.com/celo-org/optimism/pull/147/commits/c861c702f5df5394627735febf73b25203068e06

File: go.mod [hunk 2]
```diff
@@ -226,7 +226,8 @@ require (
 	rsc.io/tmplfunc v0.0.3 // indirect
 )
 
-replace github.com/ethereum/go-ethereum => github.com/celo-org/op-geth v0.0.0-20240612131021-96a2a76ceaaa
+// replace github.com/ethereum/go-ethereum => github.com/celo-org/op-geth v0.0.0-20240612131021-96a2a76ceaaa
+replace github.com/ethereum/go-ethereum => github.com/celo-org/op-geth v0.0.0-20240612164035-ef9b9a19f53e
 
 //replace github.com/ethereum/go-ethereum v1.13.9 => ../op-geth
```
Comment (piersy) [inline go.mod:232]:
```suggestion
replace github.com/ethereum/go-ethereum => github.com/celo-org/op-geth v0.0.0-20240612164035-ef9b9a19f53e
```

File: go.mod [hunk 3]
```diff
@@ -226,7 +226,8 @@ require (
 	rsc.io/tmplfunc v0.0.3 // indirect
 )
 
-replace github.com/ethereum/go-ethereum => github.com/celo-org/op-geth v0.0.0-20240612131021-96a2a76ceaaa
+// replace github.com/ethereum/go-ethereum => github.com/celo-org/op-geth v0.0.0-20240612131021-96a2a76ceaaa
+replace github.com/ethereum/go-ethereum => github.com/celo-org/op-geth v0.0.0-20240612164035-ef9b9a19f53e
```
Comment (piersy) [inline go.mod:230]: Note that we'll need to wait for my branch to be merged before you can merge this PR and after my branch has been merged you'll need to update this reference.

File: op-chain-ops/cmd/celo-dbmigrate/main.go [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+// How to run:
+// 		go run ./op-chain-ops/cmd/celo-dbmigrate -oldDB /path/to/oldDB -newDB /path/to/newDB [-batchSize 1000] [-verbosity 3] [-memoryLimit 7500] [-clear-all] [-clear-nonAncients]
+//
+// This script will migrate block data from the old database to the new database
+// You can set the log level using the -verbosity flag
+// The number of ancient records to migrate in one batch can be set using the -batchSize flag
+// The default batch size is 1000
+// You can set a memory limit in MB using the -memoryLimit flag. Defaults to 7500 MB. Make sure to set a limit that is less than your machine's available memory.
+// Use -clear-all to start with a fresh new database
+// Use -clear-nonAncients to keep migrated ancients, but not non-ancients
```
Comment (piersy) [inline op-chain-ops/cmd/celo-dbmigrate/main.go:27]: I would scrap this and move the extra info into the flag descriptions (such as for the memory limit). Also there doesn't seem to be a verbosity flag.

Comment (piersy) [reply op-chain-ops/cmd/celo-dbmigrate/main.go:27]: Also the state migration script is using `github.com/urfave/cli/v2` which produces nicer looking output and lets you provide a description of the program overall, so when we merge the two scripts I think we should use that.

File: op-chain-ops/cmd/celo-dbmigrate/main.go [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+
+func main() {
+	oldDBPath := flag.String("oldDB", "", "Path to the old database chaindata directory (read-only)")
```
Comment (piersy) [inline op-chain-ops/cmd/celo-dbmigrate/main.go:30]:
```suggestion
	oldDBPath := flag.String("oldDB", "", "Path to the old database chaindata directory (read-only) can be found at '<datadir>/celo/chaindata'")
```

File: op-chain-ops/cmd/celo-dbmigrate/main.go [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+	newDBPath := flag.String("newDB", "", "Path to the new database")
+	batchSize := flag.Uint64("batchSize", 10000, "Number of records to migrate in one batch")
+	memoryLimit := flag.Int64("memoryLimit", 7500, "Memory limit in MB")
```
Comment (piersy) [inline op-chain-ops/cmd/celo-dbmigrate/main.go:33]:
```suggestion
	memoryLimit := flag.Int64("memoryLimit", 7500, "Memory limit in MiB")
```

File: op-chain-ops/cmd/celo-dbmigrate/main.go [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+			log.Crit("Failed to remove new database", "err", err)
+		}
+	}
+	if *clearNonAncients {
+		if err := cleanupNonAncientDb(*newDBPath); err != nil {
+			log.Crit("Failed to cleanup non-ancient database", "err", err)
+		}
+	}
+
+	if err := createEmptyNewDb(*newDBPath); err != nil {
```
Comment (piersy) [inline op-chain-ops/cmd/celo-dbmigrate/main.go:70]:
```suggestion
	if err := ensureDbPath(*newDBPath); err != nil {
```

**Comment by piersy** [comment]
A couple of little points but looks fine, just we'll need to wait for PRs for support for migrated types to get merged into celo6 before we can merge this!

**Comment by alecps** [comment]
Closing in favor of https://github.com/celo-org/optimism/pull/148

**Pull request closed without merging**